File: 2022/Day1/part_two.py
import sys

def add_sorted(list: list[int], value: int, maxSize: int):
    list.append(value)
    list.sort(reverse=True)
    return list[:maxSize]

def main():
    with open(sys.argv[1], "r") as f:
        lines = f.readlines()
        maximum = []
        current = 0
        for line in lines:
            line = line.strip()
            if line == "":
                maximum = add_sorted(maximum, current, 3)
                current = 0
            else:
                current = current + int(line)
        print(sum(maximum))

if __name__ == "__main__":
    main()

File: 2022/Day1/part_two.js
const fs = require("fs");

const fileName = process.argv[2] || "data/input.txt";
const content = fs.readFileSync(fileName, "utf8");
const lines = content.split(/\r?\n/);

function include_calories(array, value, maxSize) {
  array = [...array, value].sort((a, b) => a - b);
  return array.length > maxSize ? array.slice(1) : array;
}

const result = lines.reduce(
  (acc, line) => {
    if (line === "") {
      const max = include_calories(acc.max, acc.current, 3);
      return { max, current: 0 };
    }
    const value = parseInt(line, 10);
    return { max: acc.max, current: acc.current + value };
  },
  { max: [], current: 0 }
);

const sum = result.max.reduce((a, b) => a + b, 0);

console.log(sum);
